africanlii/views/documents.py

```diff
@@ -50,11 +52,18 @@ def render_to_response(self, context, **response_kwargs):
 
 class DocumentSourcePDFView(DocumentSourceView):
     def render_to_response(self, context, **response_kwargs):
-        if hasattr(self.object, "source_file") and self.object.source_file.file:
-            if self.object.source_file.file.name.endswith(".docx"):
-                temp_dir, filename = convert_docx_to_pdf(self.object.source_file.file)
-                file = open(f"{temp_dir.name}/{filename}", "rb")
-            else:
-                file = self.object.source_file.file.open()
-            return view_attachment(file)
-        raise Http404
+        infile = self.object.source_file.file
+        if hasattr(self.object, "source_file") and infile:
+            if infile.name.endswith(".pdf"):
+                return HttpResponse(
```

Comment:
rather than copy-and-paste on the HttpResponse on this line and a few lines down, get a single file object and then have a single HttpResponse line.